lib/excon/socket.rb

```diff
@@ -364,6 +364,9 @@ def select_with_timeout(socket, type)
       if @data.include?(:deadline)
         request_timeout = request_time_remaining
 
+        # If request timeout is already reached, there's no need to proceed.
+        raise(Excon::Errors::Timeout.by_type(type, 'request')) if request_timeout <= 0
```

Comment:
I wanted this exception to be of the correct type, but instead of passing the type down to `request_time_remaining` (where this exception is raised), I figured it makes more sense to extract the raise into the caller. What do you think?

Reply:
Yeah, that seems reasonable to me. I suspect it makes it a little easier to read and understand.